src/js/features/ingestion/lastIngestion.store.ts

```diff
@@ -0,0 +1,75 @@
+import { createSlice, createAsyncThunk, PayloadAction } from '@reduxjs/toolkit';
+import axios from 'axios';
+import { katsuLastIngestionsUrl, gohanLastIngestionsUrl } from '@/constants/configConstants';
+import { printAPIError } from '@/utils/error.util';
+
+export interface DataTypeResponse {
+  count: number | null;
+  id: string;
+  label: string;
+  last_ingested: string | null;
+  queryable: boolean;
+}
+
+export type DataResponseArray = DataTypeResponse[];
+
+// Async thunks to fetch data from the two endpoints
+export const fetchKatsuData = createAsyncThunk('dataTypes/fetchKatsuData', (_, { rejectWithValue }) =>
+  axios
+    .get(katsuLastIngestionsUrl)
+    .then((res) => res.data)
+    .catch(printAPIError(rejectWithValue))
+);
+
+export const fetchGohanData = createAsyncThunk('dataTypes/fetchGohanData', (_, { rejectWithValue }) =>
+  axios
+    .get(gohanLastIngestionsUrl)
+    .then((res) => res.data)
+    .catch(printAPIError(rejectWithValue))
+);
+
+// Define the state structure
+export interface DataTypeState {
+  isFetchingData: boolean;
+  dataTypes: DataResponseArray;
+}
+
+// Initialize the state
+const initialDataTypeState: DataTypeState = {
+  isFetchingData: false,
+  dataTypes: [],
+};
+
+// Create a slice to manage the state
+const DataTypeStore = createSlice({
+  name: 'dataTypes',
+  initialState: initialDataTypeState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder.addCase(fetchKatsuData.pending, (state) => {
+      state.isFetchingData = true;
+    });
+    builder.addCase(fetchGohanData.pending, (state) => {
+      state.isFetchingData = true;
+    });
+    builder.addCase(fetchKatsuData.fulfilled, (state, { payload }: PayloadAction<DataResponseArray>) => {
+      const uniqueIds = new Set(state.dataTypes.map((data) => data.id));
+      const newData = payload.filter((data) => !uniqueIds.has(data.id));
+      state.dataTypes = [...state.dataTypes, ...newData];
```

Comment:
Missing `state.isFetchingData = false;` I think.
Also since we have standard data-types responses in katsu and gohan we can factor out the reducer in a function for both `fetchKatsuData.fulfilled` and `fetchGohanData.fulfilled`:
```typescript
const reduceServiceDataTypes = (state, { payload }: PayloadAction<DataResponseArray>) => {
      const uniqueIds = new Set(state.dataTypes.map((data) => data.id));
      const newData = payload.filter((data) => !uniqueIds.has(data.id));
      state.dataTypes = [...state.dataTypes, ...newData];
      state.isFetchingData = false;
};

// in extraReducers func
builder.addCase(fetchKatsuData.fulfilled, reduceServiceDataTypes);
builder.addCase(fetchGohanData.fulfilled, reduceServiceDataTypes);
```

Reply:
Sure! Thanks for the insigths!